Microsoft.Toolkit.Uwp.UI.Controls/RemoteDevicePicker/RemoteDevicePicker.cs

```diff
@@ -0,0 +1,210 @@
+    {
+        private Dictionary<string, RemoteSystem> DeviceMap { get; set; }
+
+        private ListView _listDevices;
+        private ComboBox _listDeviceTypes;
+        private ProgressRing _progressRing;
+
+        /// <summary>
+        /// Gets List of All Remote Systems based on Selection Filter
+        /// </summary>
+        public ObservableCollection<RemoteSystem> RemoteSystems { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the Line Color on control Header. takes **SystemControlBackgroundAccentBrush** by default
+        /// </summary>
+        public Brush HeaderLineColor
+        {
+            get { return (Brush)GetValue(HeaderLineColorProperty); }
+            set { SetValue(HeaderLineColorProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets the dependency property for <see cref="HeaderLineColor"/>.
+        /// </summary>
+        public static readonly DependencyProperty HeaderLineColorProperty = DependencyProperty.Register(
+            nameof(HeaderLineColor),
+            typeof(Brush),
+            typeof(RemoteDevicePicker),
+            new PropertyMetadata(Application.Current.Resources["SystemControlBackgroundAccentBrush"]));
+
+        /// <summary>
+        /// Gets or sets the DeviceList Selection Mode. Defaults to ListViewSelectionMode.Single
+        /// </summary>
+        public ListViewSelectionMode DeviceListSelectionMode
+        {
+            get { return (ListViewSelectionMode)GetValue(DeviceListSelectionModeProperty); }
+            set { SetValue(DeviceListSelectionModeProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets the dependency property for <see cref="DeviceListSelectionMode"/>.
+        /// </summary>
+        public static readonly DependencyProperty DeviceListSelectionModeProperty = DependencyProperty.Register(
+            nameof(DeviceListSelectionMode),
+            typeof(ListViewSelectionMode),
+            typeof(RemoteDevicePicker),
+            new PropertyMetadata(ListViewSelectionMode.Single));
+
+        /// <summary>
+        /// Fired when the Remote Device Picker is Closed.
+        /// </summary>
+        public event EventHandler<RemoteDevicePickerEventArgs> RemoteDevicePickerClosed;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RemoteDevicePicker"/> class.
+        /// </summary>
+        public RemoteDevicePicker()
+        {
+            DefaultStyleKey = typeof(RemoteDevicePicker);
+            SecondaryButtonText = "Done";
+            SecondaryButtonClick += RemoteDevicePicker_SecondaryButtonClick;
+            Loading += RemoteDevicePicker_Loading;
+            Loaded += RemoteDevicePicker_Loaded;
+        }
+
+        private void RemoteDevicePicker_SecondaryButtonClick(ContentDialog sender, ContentDialogButtonClickEventArgs args)
+        {
+            ObservableCollection<RemoteSystem> selectedItems = new ObservableCollection<RemoteSystem>();
```

Comment:
Why is an ObservableCollection needed here? a simple Collection or List should be fine

Reply:
See above. This is needed as not all RemoteSystem instances will be returned all at once. They are returned via event fired within the RemoteSystemWatcher

Reply:
No... This is the return value for being done

Reply:
True. Was thinking about a different one.